### PR TITLE
Reduce modules build in core module profile

### PR DESCRIPTION
**Pull request by aahmed-se**
Migrate pulsar cpp workflow to use the minimal module build.

### CODE REVIEW

File: .github/workflows/ci-cpp.yaml [hunk 1]
```diff
@@ -70,9 +70,9 @@ jobs:
           docker rmi $(docker images -q) -f
           df -h
 
-      - name: build pacakge
+      - name: build package
         if: steps.docs.outputs.changed_only == 'no'
-        run: mvn -B -Dorg.slf4j.simpleLogger.defaultLogLevel=ERROR package -DskipTests
+        run: mvn -B -ntp -Dorg.slf4j.simpleLogger.defaultLogLevel=ERROR -ntp install -Pcore-modules -DskipTests
```
Comment (merlimat) [inline .github/workflows/ci-cpp.yaml:75]: I think all the jobs that are running broker tests can also use just core-modules

Comment (merlimat) [reply .github/workflows/ci-cpp.yaml:75]: I mean: everything but integration tests

Comment (aahmed-se) [reply .github/workflows/ci-cpp.yaml:75]: not really we still have to run all the unit tests in these modules

Comment (merlimat) [reply .github/workflows/ci-cpp.yaml]: Yes, but unit tests are already included in broker module, there's no need to compile the entire project, just the core modules

Comment (aahmed-se) [reply .github/workflows/ci-cpp.yaml]: but no the units for io modules kafka-adaptor , slink , storm etc. I will double check after this pr which jobs need changes.

File: pom.xml [hunk 2]
```diff
@@ -1785,17 +1785,14 @@ flexible messaging model and an intuitive client API.</description>
...
         <module>pulsar-broker-common</module>
         <module>pulsar-broker</module>
-        <module>pulsar-broker-shaded</module>
         <module>pulsar-client-api</module>
         <module>pulsar-client</module>
         <module>pulsar-client-admin</module>
         <module>pulsar-client-admin-shaded</module>
         <module>pulsar-client-tools</module>
         <module>pulsar-client-tools-test</module>
-        <module>pulsar-client-all</module>
```
Comment (jiazhai) [inline pom.xml:1795]: @aahmed-se @merlimat why do we remove these modules in this file?

**Pull request closed without merging**